plugins/filter_kubernetes/kubernetes.c

```diff
@@ -786,6 +786,24 @@ static struct flb_config_map config_map[] = {
      "use 'dummy' metadata, do not talk to API server"
     },
 
+#ifdef FLB_SYSTEM_WINDOWS
+    /*
+     * Options for initial network status check on Windows pods.
+     * See fluent/fluent-bit/2144.
+     */
+    {
+     FLB_CONFIG_MAP_INT, "dns_retries", "6",
+     0, FLB_TRUE, offsetof(struct flb_kube, dns_retries),
+     "retry N times until the network starts working"
```

Comment:
"dns lookup retries N times until the network start working"

Reply:
Fixed by 8d990145.